packages/flutter/lib/src/material/action_buttons.dart

```diff
@@ -0,0 +1,419 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/widgets.dart';
+
+import 'action_icons_theme.dart';
+import 'button_style.dart';
+import 'debug.dart';
+import 'icon_button.dart';
+import 'icons.dart';
+import 'material_localizations.dart';
+import 'scaffold.dart';
+import 'theme.dart';
+
+abstract class _ActionButton extends StatelessWidget {
+  /// Creates a Material Design icon button.
+  const _ActionButton({
+    super.key,
+    this.color,
+    required this.icon,
+    required this.onPressed,
+    this.style,
+  });
+
+  /// The icon to display inside the button.
+  final Widget icon;
+
+  /// The callback that is called when the button is tapped
+  /// or otherwise activated.
+  ///
+  /// If this is set to null, the button will do a default action
+  /// when it is tapped or activated.
+  final VoidCallback? onPressed;
+
+  /// The color to use for the icon.
+  ///
+  /// Defaults to the [IconThemeData.color] specified in the ambient [IconTheme],
+  /// which usually matches the ambient [Theme]'s [ThemeData.iconTheme].
+  final Color? color;
+
+  /// Customizes this icon button's appearance.
+  ///
+  /// The [style] is only used for Material 3 [IconButton]s. If [ThemeData.useMaterial3]
+  /// is set to true, [style] is preferred for icon button customization, and any
+  /// parameters defined in [style] will override the same parameters in [IconButton].
+  ///
+  /// Null by default.
+  final ButtonStyle? style;
+
+  /// This returns the appropriate tooltip text for this action button.
+  String _getTooltip(BuildContext context);
+
+  /// This is the default function that is called when [onPressed] is set
+  /// to null.
+  void _onPressedCallback(BuildContext context);
+
+  @override
+  Widget build(BuildContext context) {
+    assert(debugCheckHasMaterialLocalizations(context));
+    return IconButton(
+      icon: icon,
+      style: style,
+      color: color,
+      tooltip: _getTooltip(context),
+      onPressed: () {
```

Comment:
This appears to be backwards-compatible with BackButton and CloseButton however it precludes creating a disabled action button.  Seems strange, but that's the way it's been, so OK.